src/Microsoft.AspNetCore.Razor.Evolution/RazorSourceDocument.cs

```diff
@@ -17,6 +17,8 @@ public abstract class RazorSourceDocument
 
         public abstract int Length { get; }
 
+        public abstract RazorSourceLineCollection Lines { get; }
```

Comment:
How would that work? You have to expose **something** that those extension methods calls. 

I don't see why this is odd. The source document is a container for text, and it also provides information about how that text is broken up into lines.